### PR TITLE
Add cmd set

### PR DESCRIPTION
**Pull request by gozeloglu**

### CODE REVIEW

File: internal/command/a.go [hunk 1]
```diff
@@ -113,12 +113,18 @@ type RHash interface {
 	Values(key string) ([]core.Value, error)
 }
 
+// RSet is a set repository.
+type RSet interface {
```
Comment (nalgeon) [inline internal/command/a.go:117]: It's better not to rush things with the `command` package. Let's deal with `rset` first.

File: internal/rset/db.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"database/sql"
+	"github.com/nalgeon/redka/internal/sqlx"
+)
+
+// DB is a database-backed set repository.
+// A set is a field-value map associated with a key.
+// Use the set repository to work with individual hashmaps
+// and their fields.
```
Comment (nalgeon) [inline internal/rset/db.go:11]: Not true for sets. It's OK to copy-paste, but please remember to tailor the code/comments to the feature you're working on.

File: internal/rset/db.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+// New connects to the set repository.
+// Does not create the database schema.
+func New(db *sql.DB) *DB {
+	d := sqlx.New(db, NewTx)
+	return &DB{d}
+}
+
+func (d *DB) Add(key string, elems ...any) (int, error) {
+	tx := NewTx(d.SQL)
+	return tx.Add(key, elems...)
```
Comment (nalgeon) [inline internal/rset/db.go:25]: `Add` modifies the database, so it should be called inside a transaction (see `rzset.DB.Add` for reference).

File: internal/rset/tx.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
+package rset
+
+import (
+	"database/sql"
+	"github.com/nalgeon/redka/internal/core"
+	"github.com/nalgeon/redka/internal/sqlx"
+	"time"
+)
+
+var sqlAdd = []string{`
```
Comment (nalgeon) [inline internal/rset/tx.go:10]: Please use consts (see `rzset` for reference).

File: internal/rset/tx.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	return &Tx{tx}
+}
+
+// Add adds key elems to set.
+func (t *Tx) Add(key string, elems ...any) (int, error) {
+	return t.add(key, elems...)
+
+}
+
+func (t *Tx) add(key string, elems ...any) (int, error) {
```
Comment (nalgeon) [inline internal/rset/tx.go:39]: What is the need for this method?

File: internal/rset/tx.go [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+	for _, arg := range args {
+		_, err := t.tx.Exec(sqlAdd[0], arg)
+		if err != nil {
+			return 0, sqlx.TypedError(err)
+		}
+		_, err = t.tx.Exec(sqlAdd[1], arg)
+		if err != nil {
+			return 0, err
+		}
+	}
```
Comment (nalgeon) [inline internal/rset/tx.go:63]: This does not really work.

File: internal/sqlx/schema.sql [hunk 7]
```diff
@@ -100,3 +100,27 @@ select
...
+-- set
+
+create table if not exists
+    rset (
+             key_id integer not null,
+             elem blob not null,
+
+             foreign key (key_id) references rkey (id)
+    on delete cascade
+    );
```
Comment (nalgeon) [inline internal/sqlx/schema.sql:113]: Please fix the indentation.

File: internal/sqlx/schema.sql [hunk 8]
```diff
@@ -100,3 +100,27 @@ select
...
+
+create unique index if not exists
+    rset_pk_idx on rset (key_id, elem);
+
+create view if not exists
+        vset as
+select
+    rkey.id as key_id, rkey.key, rset.elem
+               datetime(etime/1000, 'unixepoch') as etime,
+        datetime(mtime/1000, 'unixepoch') as mtime
```
Comment (nalgeon) [inline internal/sqlx/schema.sql:123]: Also indentation problems.

File: internal/command/a.go (hunk 1 above)
Comment (gozeloglu) [reply internal/command/a.go:117]: Shouldn't I implement the `internal/command` part?

File: internal/rset/db.go (hunk 2 above)
Comment (gozeloglu) [reply internal/rset/db.go:11]: Yes, I know. This PR is not ready yet. I am going to update it. 👍🏻

File: internal/rset/tx.go (hunk 5 above)
Comment (gozeloglu) [reply internal/rset/tx.go:39]: I thought that we may need to reuse the same method for other commands, but I might be overengineering. I can remove it.

File: internal/rset/tx.go (hunk 4 above)
Comment (gozeloglu) [reply internal/rset/tx.go:10]: OK. I saw a slice of queries in another command. I referred it from there, but `sqlAdd1` and `sqlAdd2` make sense.

**Pull request closed without merging**